lib/vtls/schannel.c

```diff
@@ -1974,7 +2112,7 @@ schannel_recv(struct Curl_easy *data, int sockindex,
           infof(data, "schannel: can't renegotiate, an error is pending");
           goto cleanup;
         }
-        if(BACKEND->encdata_offset) {
+        if(BACKEND->encdata_offset && inbuf[1].cbBuffer) {
```

Comment:
This change seems unrelated to TLS 1.3.

Reply:
It is required. I spend a few days mulling over and debugging what exactly was happening. With the new SChannel support of `SCH_CREDENTIALS`, inbuf[1] can be empty while `encdata_offset` is not empty when receiving a `SEC_I_RENEGOTIATE` return code.

Unfortunately, MS's documentation on this behavior is slim, so I can't tell you *why* this is the case, just that it *is* the case.

Long story short: without this change `SCH_CREDENTIALS` support (and thus TLS 1.3 support) will fail.

Reply:
Understood, but what happens for applications not using TLS 1.3 following the old code path?

Reply:
Same working behavior when using TLS 1.2 (or older) on either the `SCH_CREDENTIALS` or `SCHANNEL_CRED` code-paths.

Reply:
It looks like this change effectively renders the check moot. I added the check in 3e7ec1e, but going through it just now I can't remember why. Maybe I thought it was not possible that encrypted data would be sent before the renegotiate completed? However my note in the commit message contradicts that guess:

~~~
- Do not recurse on renegotiation.

Instead we'll continue on to process any trailing encrypted data
received during the renegotiation only.
~~~

So I think my intention was to process that data.

Reply:
Unfortunately I couldn't reproduce a case where `SEC_I_RENEGOTIATE` was received, `BACKEND->encdata_offset` was `!= 0` and it wasn't just faulty error handling. The code *still* handle the case where `inbuf[1].cbBuffer == 0`  and `inbuf[3].cbBuffer != 0`. But is that necessary (or is it a code-path that will never be executed)? I have no idea -- I couldn't reproduce a case that got that result.

Unfortunately that commit message and those PR discussions didn't get me any closer to reproducing a test-case.